app/assets/javascripts/components/uploads/upload_list.jsx

```diff
@@ -62,31 +65,34 @@ const UploadList = createReactClass({
     let uploadsView;
 
     if (this.props.view === GALLERY_VIEW) {
-      uploadsView = (
-        <div className="gallery-view">
-          {elements}
-        </div>
-      );
+      if (elements.length === 0) {
+        uploadsView = (<div className="list-view">{noUploadsMessage}</div>);
+      } else {
+        uploadsView = (<div className="gallery-view"> {elements} </div>);
+      }
     }
 
     if (this.props.view === LIST_VIEW) {
-      uploadsView = (
-        <div className="list-view">
-          <List
-            elements={elements}
-            keys={keys}
-            table_key="uploads"
-            sortBy={this.props.sortBy}
-            none_message={I18n.t('courses_generic.uploads_none')}
-          />
-        </div>
-      );
+      if (elements.length && elements.length > 0) {
```

Comment:
Are both of these checks necessary? I'd think either one would be sufficient. I can't think of a way that `elements.length` would be truthy but not greater than 0.

Reply:
oh, got it. I thought it will be undefined if elements is not an array